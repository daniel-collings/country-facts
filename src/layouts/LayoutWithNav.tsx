import { Link } from 'react-router-dom'
import { useTheme } from '@/hooks/useTheme.tsx'
import { ThemeEnum } from '@/types/enums.ts'

export default function LayoutWithNav({
  children
}: {
  children: React.ReactNode
}) {
  const { setTheme } = useTheme()

  return (
    <div className="space-y-4">
      <div className="navbar bg-base-200 rounded-xl">
        <div className="flex-1 px-4">
          <Link to="/" className="text-xl">
            Country Facts
          </Link>
        </div>
        <div className="flex-none">
          <ul className="menu menu-horizontal px-1">
            <li>
              <details>
                <summary>Theme</summary>
                <ul className="bg-base-100 rounded-t-none p-2 max-h-56 overflow-auto">
                  {Object.keys(ThemeEnum).map((theme) => (
                    <li key={theme}>
                      <button onClick={() => setTheme(ThemeEnum[theme])}>
                        {ThemeEnum[theme]}
                      </button>
                    </li>
                  ))}
                </ul>
              </details>
            </li>
            <li>
              <Link to="/countries">Countries</Link>
            </li>
          </ul>
        </div>
      </div>
      {children}
    </div>
  )
}
