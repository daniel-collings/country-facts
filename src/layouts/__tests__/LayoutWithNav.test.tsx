import { describe, expect, test } from 'vitest'
import { render } from '@/utils/testUtils.tsx'
import LayoutWithNav from '@/layouts/LayoutWithNav.tsx'

describe('LayoutWithNav Component', () => {
  test('should render correctly', () => {
    const { asFragment } = render(
      <LayoutWithNav>
        <h1>Hello World</h1>
      </LayoutWithNav>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
