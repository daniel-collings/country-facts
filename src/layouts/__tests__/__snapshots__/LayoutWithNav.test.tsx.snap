// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LayoutWithNav Component > should render correctly 1`] = `
<DocumentFragment>
  <div
    class="h-full min-h-screen bg-base-100 p-4 sm:p-8"
    data-theme="light"
  >
    <main
      class="max-w-7xl mx-auto px-0 sm:px-6 lg:px-8"
    >
      <div
        class="space-y-4"
      >
        <div
          class="navbar bg-base-200 rounded-xl"
        >
          <div
            class="flex-1 px-4"
          >
            <a
              class="text-xl"
              href="/"
            >
              Country Facts
            </a>
          </div>
          <div
            class="flex-none"
          >
            <ul
              class="menu menu-horizontal px-1"
            >
              <li>
                <details>
                  <summary>
                    Theme
                  </summary>
                  <ul
                    class="bg-base-100 rounded-t-none p-2 max-h-56 overflow-auto"
                  >
                    <li>
                      <button>
                        light
                      </button>
                    </li>
                    <li>
                      <button>
                        dark
                      </button>
                    </li>
                    <li>
                      <button>
                        cupcake
                      </button>
                    </li>
                    <li>
                      <button>
                        bumblebee
                      </button>
                    </li>
                    <li>
                      <button>
                        emerald
                      </button>
                    </li>
                    <li>
                      <button>
                        corporate
                      </button>
                    </li>
                    <li>
                      <button>
                        synthwave
                      </button>
                    </li>
                    <li>
                      <button>
                        retro
                      </button>
                    </li>
                    <li>
                      <button>
                        cyberpunk
                      </button>
                    </li>
                    <li>
                      <button>
                        valentine
                      </button>
                    </li>
                    <li>
                      <button>
                        halloween
                      </button>
                    </li>
                    <li>
                      <button>
                        garden
                      </button>
                    </li>
                    <li>
                      <button>
                        forest
                      </button>
                    </li>
                    <li>
                      <button>
                        aqua
                      </button>
                    </li>
                    <li>
                      <button>
                        lofi
                      </button>
                    </li>
                    <li>
                      <button>
                        pastel
                      </button>
                    </li>
                    <li>
                      <button>
                        fantasy
                      </button>
                    </li>
                    <li>
                      <button>
                        wireframe
                      </button>
                    </li>
                    <li>
                      <button>
                        black
                      </button>
                    </li>
                    <li>
                      <button>
                        luxury
                      </button>
                    </li>
                    <li>
                      <button>
                        dracula
                      </button>
                    </li>
                    <li>
                      <button>
                        cmyk
                      </button>
                    </li>
                    <li>
                      <button>
                        autumn
                      </button>
                    </li>
                    <li>
                      <button>
                        business
                      </button>
                    </li>
                    <li>
                      <button>
                        acid
                      </button>
                    </li>
                    <li>
                      <button>
                        lemonade
                      </button>
                    </li>
                    <li>
                      <button>
                        night
                      </button>
                    </li>
                    <li>
                      <button>
                        coffee
                      </button>
                    </li>
                    <li>
                      <button>
                        winter
                      </button>
                    </li>
                    <li>
                      <button>
                        dim
                      </button>
                    </li>
                    <li>
                      <button>
                        nord
                      </button>
                    </li>
                    <li>
                      <button>
                        sunset
                      </button>
                    </li>
                  </ul>
                </details>
              </li>
              <li>
                <a
                  href="/countries"
                >
                  Countries
                </a>
              </li>
            </ul>
          </div>
        </div>
        <h1>
          Hello World
        </h1>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
